go/main.go

```diff
@@ -0,0 +1,140 @@
+package main
+
+import (
+	"fmt"
+	"os"
+
+	"github.com/dgrijalva/jwt-go"
+	"github.com/gofiber/fiber"
+	"github.com/gofiber/template/pug"
+	"golang.org/x/oauth2"
+)
+
+/* #nosec G1010*/
+const (
+	accessTokenURL   = "https://api.creditkudos-staging.com/v2/oauth/token"
+	authorizationURL = "https://auth.creditkudos-staging.com/oauth/authorize"
+)
+
+// Update the env in docker with your redirect URL and client id/secret
+var (
+	redirectURL  = ""
+	clientID     = ""
+	clientSecret = ""
+)
+
+func main() {
+	redirectURL = os.Getenv("REDIRECTURL")
+	clientID = os.Getenv("CLIENTID")
+	clientSecret = os.Getenv("CLIENTSECRET")
+	if redirectURL == "" || clientID == "" || clientSecret == "" {
+		panic("Set env variables before running")
+	}
+
+	engine := pug.New("./views", ".pug")
+	app := fiber.New(&fiber.Settings{Views: engine})
+	oauthClient := createOauthClient()
+
+	createRoutes(app, oauthClient)
+
+	if err := app.Listen(8080); err != nil {
+		fmt.Println("App errored: ", err.Error())
+	}
+}
+
+func createOauthClient() *oauth2.Config {
+	return &oauth2.Config{
+		ClientID:     clientID,
+		ClientSecret: clientSecret,
+		Scopes:       []string{"read"},
+		Endpoint: oauth2.Endpoint{
+			AuthURL:  authorizationURL,
+			TokenURL: accessTokenURL,
+		},
+		RedirectURL: redirectURL,
+	}
+}
+
+func createRoutes(app *fiber.App, oauthConfig *oauth2.Config) {
+	// Setup landing page
+	app.Get("/", func(c *fiber.Ctx) {
+		if err := c.Render(
+			"index",
+			fiber.Map{
+				"title":   "Example Go implementation",
+				"message": "Click below to start a journey",
+			},
+		); err != nil {
+			fmt.Println("Failed to load main page:", err)
```

Comment:
`c.SendStatus(http.StatusInternalServerError)` too?

Reply:
Changed to use `handleError`, will add this to `handleError` on fatal.